### ISSUE TITLE
azurerm_kubernetes_cluster: add support for enableNodePublicIP

### ISSUE THREAD

**Issue by iameli** [terraform-providers/terraform-provider-azurerm#4581]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure has added support for an `enableNodePublicIP` field in the [AgentPoolProfile schema](https://docs.microsoft.com/en-us/azure/templates/microsoft.containerservice/2019-06-01/managedclusters#ContainerServiceNetworkProfile). Would be nice to be able to make use of it from Terraform.

### New or Affected Resource(s)

* azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster" "test" {
  name                = "test"
  location            = "UK South"
  resource_group_name = "testrg"
  dns_prefix          = "test"

  agent_pool_profile {
    name                  = "poola"
    count                 = 2
    vm_size               = "Standard_D8_v3"
    os_type               = "Linux"
    os_disk_size_gb       = 256
    type                  = "VirtualMachineScaleSets"
    enable_node_public_ip = true
  }

  agent_pool_profile {
    name                  = "poolb"
    count                 = 1
    vm_size               = "Standard_D8_v3"
    os_type               = "Linux"
    os_disk_size_gb       = 256
    type                  = "VirtualMachineScaleSets"
    enable_node_public_ip = false
  }
}
```

I'm going to need something like this within the next couple of months, so I'll implement it then unless someone else beats me to it.

**Comment by topikettunen** [comment]
```
resource "azurerm_public_ip" "nginx_ingress" {
  name                = "nginx-ingress-pip"
  location            = azurerm_kubernetes_cluster.aks.location
  resource_group_name = azurerm_kubernetes_cluster.aks.node_resource_group
  allocation_method   = "Static"
  domain_name_label   = "cool-label-k8s" 
}
```

I've used something similar to this when I've needed public IP for example when working with ingress. Is this kind of approach sufficient enough until this enhancement is done?

**Comment by topikettunen** [comment]
I could tackle this!

Currently we've used following approach when working with public IPs and, for example, AKS ingress:

```
resource "azurerm_public_ip" "nginx_ingress" {
  name                = "nginx-ingress-pip"
  location            = azurerm_kubernetes_cluster.aks.location
  resource_group_name = azurerm_kubernetes_cluster.aks.node_resource_group
  allocation_method   = "Static"
  domain_name_label   = "cool-domain-k8s" 
}
```

and personally I believe that enabling public IP simply in agent pool profile would result to a slightly cleaner approach.

**Comment by jluk** [comment]
FWIW this property is captured under issue #4001 which should exist as an agent pool profile property. Also for awareness this capability is still in preview so that's a good callout if it does get brought in.

**Comment by tombuildsstuff** [comment]
hey @iameli 

@topikettunen has added support for this in #4613 which has merged and will ship as a part of 1.37 - as such I'm going to close this issue for the moment. Taking a look through it appears the documentation for this field is missing, but I've got a fix for this locally which I'll push with a related PR later today.

Thanks!
